web-nfc.d.ts

```diff
@@ -22,9 +22,11 @@ declare type NDEFRecordDataSource = string | BufferSource | NDEFMessageInit
 interface Window {
   NDEFRecord: NDEFRecord
 }
+type NDEFRecordTypes = string | "absolute-url" | "empty" | "mime" | "smart-poster" | "text" | "unknown" | "URL";
```

Comment:
It is incorrect to include `string` here. In this definition, it is synonymous with `string`.
If exclude `string`, we will get the expected compilation error. [What this means](https://www.typescriptlang.org/play?#code/C4TwDgpgBA+gcgEQKIDEBKEDGB7ATgEwBVwIBnKAXilOFwEsA7AcygB8oAiAQwCNTsANgFdgEALRDcAjm04QAtmFAz2HeXXkQVnUvK65gYsNhoRc2jqIAewC0IYBrBtgDuDCwFU0AGQ4BuAFgAKGCAelCoQgALM2g6cmcoM1w8YJwGGmouTQBBUgBlWkYmAC5YRFQMHAJiSHIqAHJ1Gi4HCAbg4NBIKAr0LDwiEnrOXn5hUQkpCwUlEAt1TQtdfUNjU3NZSwgbO0dnN08ff06g8MiY3DjyaqvMYCTcFNw07AyHnch7iHwkJ7wyn0qoNamRKFAmvFgK12sEgA)

Union types of such `string literal type` should be used to avoid including incorrect strings.

Reply:
Yes, if we want a language level (early) error. Just noting that it's also possible to defer that decision (to accept or not a given string record type) to implementations: then a general string should be fine, including `"mistake"`. It depends how do we want to do it.
IMHO we (Web NFC) are not in the position to extend the range of valid NFC record types or leave extension possibilities to implementations, therefore I agree on using string literals.

Reply:
It is correct that "NDEF Record Type" can specify any character string (local type and external type), So it should be a `string`.Committing as a document is good, but it doesn't contribute to Type Safe.

Reply:
I just went ahead and updated it to include the declare keyword and also removed the string type.